src/components/ui/combobox.tsx

```diff
@@ -57,6 +58,7 @@ interface ComboBoxProps extends React.PropsWithChildren {
   max?: number
   sortOrder?: 'label' | 'value'
   modal?: boolean
+  fetchAsyncOptions?: (query: string) => Promise<DefaultValueOption[]>
```

Comment:
This should probably be renamed (plain `fetch`?) and typed as a generic, we want to be able to fetch all types of data.